packages/edit-site/src/components/global-styles/font-library-modal/utils/preview-styles.js

```diff
@@ -30,22 +30,81 @@ function extractFontWeights( fontFaces ) {
+ * It follows the recommendations from the CSS Fonts Module Level 4.
+ * https://www.w3.org/TR/css-fonts-4/#font-family-prop
+ *
+ * @param {string} input - The font family.
+ * @return {string} The formatted font family.
+ *
+ * Example:
+ * formatFontFamily( "Open Sans, Font+Name, sans-serif" ) => '"Open Sans", "Font+Name", sans-serif'
+ * formatFontFamily( "'Open Sans', sans-serif" ) => '"Open Sans", sans-serif'
+ * formatFontFamily( "DotGothic16, Slabo 27px, serif" ) => '"DotGothic16","Slabo 27px",serif'
+ * formatFontFamily( "Mine's, Moe's Typography" ) => `"mine's","Moe's Typography"`
+ */
 export function formatFontFamily( input ) {
-	return input
-		.split( ',' )
-		.map( ( font ) => {
-			font = font.trim(); // Remove any leading or trailing white spaces
-			// If the font doesn't start with quotes and contains a space, then wrap in quotes.
-			// Check that string starts with a single or double quote and not a space
-			if (
-				! ( font.startsWith( '"' ) || font.startsWith( "'" ) ) &&
-				font.indexOf( ' ' ) !== -1
-			) {
-				return `"${ font }"`;
-			}
-			return font; // Return font as is if no transformation is needed
-		} )
-		.join( ', ' );
+	// Matchs any non alphabetic characters (a-zA-Z), dashes - , or parenthesis ()
+	const regex = /[^a-zA-Z\-()]+/;
+	const output = input.trim();
+
+	const formatItem = ( item ) => {
+		item = item.trim();
+		if ( item.match( regex ) ) {
+			// removes leading and trailing quotes.
+			item = item.replace( /^["']|["']$/g, '' );
+			return `"${ item }"`;
+		}
+		return item;
+	};
+
+	if ( output.includes( ',' ) ) {
+		return output
+			.split( ',' )
+			.map( formatItem )
+			.filter( ( item ) => item !== '' )
+			.join( ', ' );
+	}
+
+	return formatItem( output );
+}
+
+/*
+ * Format the font face name to use in the font-family property of a font face.
+ *
+ * The input can be a string with the font face name or a string with multiple font face names separated by commas.
+ * It removes the leading and trailing quotes from the font face name.
+ *
+ * @param {string} input - The font face name.
+ * @return {string} The formatted font face name.
+ *
+ * Example:
+ * formatFontFaceName("Open Sans") => "Open Sans"
+ * formatFontFaceName("'Open Sans', sans-serif") => "Open Sans"
+ * formatFontFaceName(", 'Open Sans', 'Helvetica Neue', sans-serif") => "Open Sans"
+ */
+export function formatFontFaceName( input ) {
+	let output = input.trim();
+	if ( output.includes( ',' ) ) {
+		output = output
+			.split( ',' )
+			// finds the first item that is not an empty string.
+			.find( ( item ) => item.trim() !== '' )
+			.trim()
+			// removes leading and trailing quotes.
+			.replace( /^["']|["']$/g, '' );
+	}
+	// removes leading and trailing quotes.
+	output = output.replace( /^["']|["']$/g, '' );
+
+	// Firefox needs the font name to be wrapped in double quotes meanwhile other browsers don't.
+	if ( window.navigator.userAgent.toLowerCase().includes( 'firefox' ) ) {
```

Comment:
Nit: Should we check for `gecko` to make it more broad and cover not only Firefox but all gecko browsers?

Reply:
I tried with 'gecko' https://github.com/WordPress/gutenberg/pull/59019/commits/2f3a0dfabcaae6e3a8b33a4d119f2aff33b23f30, but after that, I noticed that "Gecko" string is in a wide range of user-agents, so I think we should use "firefox" and "fxios"(firefox for ios). Added that here: https://github.com/WordPress/gutenberg/pull/59019/commits/f85650a927669ebb4da74b2e43680b08064f623a

Reference:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
https://deviceatlas.com/blog/list-of-user-agent-strings

Reply:
Note that Firefox on iOS uses the Safari/Webkit engine (as do all iOS browsers, as currently required by Apple), so I'm guessing it will need the same treatment as Safari (no quotes), I'll see if I can test it.

Reply:
Good call; I didn't know that. I'm unable to test on iOS, but if it's not Gecko engine, we should not include it among the exceptions. I removed that from the conditional in this PR: https://github.com/WordPress/gutenberg/pull/59037

Reference: https://firefox-source-docs.mozilla.org/overview/ios.html